### PR TITLE
Improved error message for break in async block

### LINKED ISSUE #63391: Improve diagnostic for `break` in `async { ... }` block
With:
```rust
#![feature(async_await)]

fn foo() {
    async {
        break;
    };
}
```
you get:
```rust
error[E0267]: `break` inside of a closure
 --> src/lib.rs:5:9
  |
5 |         break;
  |         ^^^^^ cannot break inside of a closure
```

Note how it references "a closure" which seems confusing for a user who doesn't presume the implementation.

### PR DESCRIPTION
**Pull request by gilescope**
Fixes #63391

### CODE REVIEW

File: src/librustc_passes/loops.rs [hunk 1]
```diff
@@ -57,6 +58,11 @@ impl<'a, 'hir> Visitor<'hir> for CheckLoopVisitor<'a, 'hir> {
             hir::ExprKind::Loop(ref b, _, source) => {
                 self.with_context(Loop(source), |v| v.visit_block(&b));
             }
+            hir::ExprKind::Closure(_, ref function_decl, b, _,
+                                   Some(GeneratorMovability::Static)) => {
```
Comment (Centril) [inline src/librustc_passes/loops.rs]: I'd prefer using a `match` here on `movability` so that the non-diagnostic logic isn't repeated.

**Comment by Centril** [comment]
r? @Centril 

r=me rollup with comment addressed.

File: src/librustc_passes/loops.rs (hunk 1 above)
Comment (gilescope) [reply src/librustc_passes/loops.rs]: Good spot

**Pull request merged**